tests/functional/diff/test_diff.py

```diff
@@ -299,7 +299,7 @@ def test_get_diff_checker_counts_core_unresolved(self):
                                                     cmp_data)
         # Unesolved core checkers.
         test_res = {'core.NullDereference': 4, 'core.DivideZero': 3}
-        self.assertDictEqual(diff_res, test_res)
+        self.assertDictContainsSubset(test_res, diff_res)
```

Comment:
I would add some comments here, how the filtering behavior changed and why a subset check is enough or we should compare to the whole result returned in `diff_res`  as before and change `test_res`. 

This comment applies to the other tests too which were changed.